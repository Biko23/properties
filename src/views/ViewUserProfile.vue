<template>
<div>
    <v-container>
        <v-row>
            <v-col cols="12" sm="12" md="8" xl="8" offset-lg="2" offset-md="2">
                <h3 style="
                    color: #3b6ef3;
                    font-weight: 400;
                    text-align: center;
                    margin-top: 20px;
                ">Settings</h3>
                <br />
                <v-container 
                style="
                    background-color: rgb(231, 240, 255);
                    padding: 0 20%;
                    border-radius: 1%;
                    "
                >
                    <v-row>
                        <v-col cols="12" sm="12" md="5">
                            <v-text>Full Name:</v-text>
                        </v-col>
                        <v-col cols="12" sm="12" md="7">
                            <v-text>{{currentLoggedinUser.vendor_name}}</v-text>
                        </v-col>
                    </v-row>
                    <v-row>
                        <v-col cols="12" sm="12" md="5">
                            <v-text>Username:</v-text>
                        </v-col>
                        <v-col cols="12" sm="12" md="7">
                            <v-text>{{currentLoggedinUser.username}}</v-text>
                        </v-col>
                    </v-row>
                    <v-row>
                        <v-col cols="12" sm="12" md="5">
                            <v-text>Primary Phone:</v-text>
                        </v-col>
                        <v-col cols="12" sm="12" md="7">
                            <v-text>{{currentLoggedinUser.vendor_primary_phone_number}}</v-text>
                        </v-col>
                    </v-row>
                    <v-row>
                        <v-col cols="12" sm="12" md="5">
                            <v-text>Primary Email:</v-text>
                        </v-col>
                        <v-col cols="12" sm="12" md="7">
                            <v-text>{{currentLoggedinUser.vendor_primary_email}}</v-text>
                        </v-col>
                    </v-row>
                    <v-row>
                        <v-col cols="12" sm="12" md="5">
                            <v-text>Secondary Phone:</v-text>
                        </v-col>
                        <v-col cols="12" sm="12" md="7">
                            <v-text>{{currentLoggedinUser.vendor_secondary_phone_number}}</v-text>
                        </v-col>
                    </v-row>
                    <v-row>
                        <v-col cols="12" sm="12" md="5">
                            <v-text>Secondary Email:</v-text>
                        </v-col>
                        <v-col cols="12" sm="12" md="7">
                            <v-text>{{currentLoggedinUser.vendor_secondary_email}}</v-text>
                        </v-col>
                    </v-row>
                     <v-row>
                        <v-col cols="12" sm="12" md="5">
                            <v-text>Location:</v-text>
                        </v-col>
                        <v-col cols="12" sm="12" md="7">
                            <v-text>{{currentLoggedinUser.business_location}}</v-text>
                        </v-col>
                    </v-row>
                    <v-row>
                        <v-col cols="12" sm="12" md="12">
                            <router-link to="/editprofile" style="text-decoration:none;">
                                <v-btn depressed color="#3b6ef3" block>
                                    <span style="color: white;">Edit Profile</span></v-btn>
                            </router-link>
                        </v-col>
                    </v-row>
                </v-container>
            </v-col>
        </v-row>
    </v-container><br><br>
</div>
</template>
<script>
import {
    mapGetters
} from 'vuex';

export default {
    name: 'Home',
    computed: {
        ...mapGetters(["currentLoggedinUser"]),
    }
}
</script>
