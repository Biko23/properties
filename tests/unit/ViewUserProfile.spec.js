import { createLocalVue, mount } from '@vue/test-utils'
import ViewUserProfile from '@/views/ViewUserProfile.vue'
import Vuetify from 'vuetify'
import Vuex from 'vuex'
import VueRouter from 'vue-router'


describe('ViewUserProfile.vue', () => {
    const localVue = createLocalVue()
    localVue.use(Vuex)
    localVue.use(VueRouter)
    let actions
    let getters
    let store
    let vuetify
    let router

    beforeEach(() => {
        vuetify = new Vuetify()
        router = new VueRouter()
        getters = {
            currentLoggedinUser: () => ({})
        }
        store = new Vuex.Store({
            getters
        })
    })

    const mountFactory = options => {
        return mount(ViewUserProfile, {
            localVue,
            vuetify,
            store,
            router,
            ...options
        })
    }

    it('should have a match snapshot', () => {
        const wrapper = mountFactory({})
        expect(wrapper.html()).toMatchSnapshot()
    })
})

