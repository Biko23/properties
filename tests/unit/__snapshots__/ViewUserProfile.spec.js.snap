// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewUserProfile.vue should have a match snapshot 1`] = `
<div>
  <div class="container">
    <div class="row">
      <div class="col-sm-12 col-md-8 col-xl-8 offset-md-2 offset-lg-2 col-12">
        <h3 style="color: rgb(59, 110, 243); font-weight: 400; text-align: center; margin-top: 20px;">Settings</h3> <br>
        <div class="container" style="background-color: rgb(231, 240, 255); padding: 0px 20%; border-radius: 1%;">
          <div class="row">
            <div class="col-sm-12 col-md-5 col-12"><span>Full Name:</span></div>
            <div class="col-sm-12 col-md-7 col-12"><span></span></div>
          </div>
          <div class="row">
            <div class="col-sm-12 col-md-5 col-12"><span>Username:</span></div>
            <div class="col-sm-12 col-md-7 col-12"><span></span></div>
          </div>
          <div class="row">
            <div class="col-sm-12 col-md-5 col-12"><span>Primary Phone:</span></div>
            <div class="col-sm-12 col-md-7 col-12"><span></span></div>
          </div>
          <div class="row">
            <div class="col-sm-12 col-md-5 col-12"><span>Primary Email:</span></div>
            <div class="col-sm-12 col-md-7 col-12"><span></span></div>
          </div>
          <div class="row">
            <div class="col-sm-12 col-md-5 col-12"><span>Secondary Phone:</span></div>
            <div class="col-sm-12 col-md-7 col-12"><span></span></div>
          </div>
          <div class="row">
            <div class="col-sm-12 col-md-5 col-12"><span>Secondary Email:</span></div>
            <div class="col-sm-12 col-md-7 col-12"><span></span></div>
          </div>
          <div class="row">
            <div class="col-sm-12 col-md-5 col-12"><span>Location:</span></div>
            <div class="col-sm-12 col-md-7 col-12"><span></span></div>
          </div>
          <div class="row">
            <div class="col-sm-12 col-md-12 col-12"><a href="#/editprofile" class="" style="text-decoration: none;"><button type="button" class="v-btn v-btn--block v-btn--has-bg theme--light v-size--default" style="background-color: rgb(59, 110, 243); border-color: #3b6ef3;"><span class="v-btn__content"><span style="color: white;">Edit Profile</span></span></button></a></div>
          </div>
        </div>
      </div>
    </div>
  </div><br><br>
</div>
`;
